### PR TITLE
More robust heap initialization w/LiveUpdate and SystemLog

### PR DESCRIPTION
**Pull request by fwsGonzo**

### CODE REVIEW

File: src/hal/machine.cpp [hunk 1]
```diff
@@ -89,6 +89,18 @@ namespace os::detail {
     MINFO("Initializing heap\n");
     auto main_mem = memory().allocate_largest();
     MINFO("Main memory detected as %zu b\n", main_mem.size);
+    memory().deallocate(main_mem.ptr, main_mem.size);
+
+#ifndef PLATFORM_x86_solo5
```
Comment (KristianJerpetjon) [inline src/hal/machine.cpp]: We could use something more generic from cmake ie PLATFORM_SUPPORTS_LIVEUPDATE ?

File: src/hal/machine.cpp [hunk 2]
```diff
@@ -89,6 +89,18 @@ namespace os::detail {
...
+    static const size_t SYSTEMLOG_SIZE = 65536;
+    const size_t LIU_SIZE = (main_mem.size / (100 / 25)) & ~0xfff;
```
Comment (KristianJerpetjon) [inline src/hal/machine.cpp]: the last part is to make sure its rounded down to 4k ? so you get somewhat less than 25% of all ram

File: lib/LiveUpdate/src/os.cpp [hunk 3]
```diff
@@ -3,47 +3,43 @@
...
-  return phys_max / (100 / size_t(LIVEUPDATE_AREA_SIZE));
-}
-uintptr_t kernel::liveupdate_phys_loc(size_t phys_max)  noexcept {
-  return phys_max & ~(uintptr_t) 0xFFF;
-}
-
 void kernel::setup_liveupdate(uintptr_t phys)
 {
-  static uintptr_t temp_phys = 0;
+  static uintptr_t temp_phys;
```
Comment (alfred-bratterud) [inline lib/LiveUpdate/src/os.cpp]: Explicitly removing zero-initialization. Is there a better way to solve this?

File: src/hal/machine.cpp [hunk 4]
```diff
@@ -89,6 +89,18 @@ namespace os::detail {
     MINFO("Initializing heap\n");
     auto main_mem = memory().allocate_largest();
     MINFO("Main memory detected as %zu b\n", main_mem.size);
+    memory().deallocate(main_mem.ptr, main_mem.size);
```
[the remark this answers is not in the document]
Comment (fwsGonzo) [reply src/hal/machine.cpp]: That was essentially what you did before, I just moved it

**Pull request merged**